src/h5web/dataset-visualizer/VisProvider.tsx

```diff
@@ -50,4 +51,10 @@ export function useVisProps(): VisProps {
   return props;
 }
 
+export function useFlatValues(): number[] {
+  const { rawValues } = useVisProps();
+
+  return useMemo(() => rawValues.flat(Infinity), [rawValues]);
+}
+
```

Comment:
A new hook to get the values *completely* flattened to slice deep dimensions. Otherwise, `rawValues.flat()` only flattens the first level.

WE might want to replace the `useValues` in `HeatmapVis` by this.